### PR TITLE
Makefile refactor

### LINKED ISSUE #206: Docker-compose trying to pull non-existant tag for jupyterhub
Command "PLATFORM_VERSION=development docker-compose pull"
 
Results in 
"Pulling singleuser (rengahub/singleuser:development)...
ERROR: manifest for rengahub/singleuser:development not found"

After which pull does not continue.

### PR DESCRIPTION
**Pull request by rokroskar**
Major change is the removal of deployment-related pieces from the `Makefile`. These are moved to two scripts in the `scripts` directory instead. Documentation (READMEs and the docs) has been adapted accordingly. 

addresses #149 #186 
closes #206

### CODE REVIEW

**Comment by leafty** [comment]
the way this works now, i would also remove the `.env` target

**Comment by rokroskar** [comment]
@leafty you still need some way to remember those environment variables

**Comment by rokroskar** [comment]
However, I might soon advocate for removing the docker-compose and only using helm. It's getting pretty ridiculous trying to maintain the two set up and helm seems now less error prone.

**Comment by leafty** [comment]
maybe i'm wrong, but i thought only the `$PLATFORM_VERSION` was used for builds. If that's correct, it's way simpler to have it as a regular env var.

the other vars are part of the start script for me, so that's why i asked about moving the target to some part of the script.

**Comment by rokroskar** [comment]
@leafty if you run the script twice and you change the token values between runs, the registration/unregistration of various pieces will fail. That's why we used the `.env` file in the first place.

**Comment by leafty** [comment]
not denying that. I'm saying that creating/managing the `.env` file should happen in the scripts, not in the Makefile.

**Comment by jirikuncar** [comment]
> I'm saying that creating/managing the `.env` file should happen in the scripts, not in the Makefile.

The `.env` file is a generated file, hence it's a make target.

**Comment by erbou** [comment]
I could test the branch and run the demo. In the make start I got the error:

Checking that services are reachable...
make: *** [start] Error 2

That's because *.renga.build did not respond to the ping command at the end of the renga-start.sh. Curl is more heavy handed than ping, but since it works we should use that.

There is an issue in renga-wipe.sh, where the script continues even if .env does not exist.

**Comment by rokroskar** [comment]
I don't understand, so ping fails somehow but curl works?

**Comment by erbou** [comment]
The local host does not respond to ICMP, the ping command hangs and timeouts with 100% packet loss:

>>>> ping -c1 renga.build
PING keycloak.renga.build (127.0.0.1): 56 data bytes
--- keycloak.renga.build ping statistics ---
1 packets transmitted, 0 packets received, 100.0% packet loss

**Comment by rokroskar** [comment]
is this something you have specifically disabled? or is this normal? I've never seen that on my machine.

**Comment by erbou** [comment]
The firewall on my macbook is on (always). By default it does not filter TCP/UDP connections coming from my localhost, but it does filter ICMP. It is not unusual for IT departments to check or impose that their employees have their Firewall enabled, and it wouldn't hurt to support that.

**Comment by rokroskar** [comment]
Yep makes sense will fix it

File: Makefile [hunk 1]
```diff
@@ -132,10 +197,12 @@ scala-artifact = \
 	renga-graph
```
Comment (leafty) [inline Makefile]: TODO: remove the scala artifacts and update readme to remove sbt as a req.

NB: `renku-storage` cab be built with docker only. Also, cascading changes from `renku-commons` is not possible from the docker build so building it here has no value.

**Comment by rokroskar** [comment]
@leafty an error in the singleuser base image crept in while I was merging in the changes from master -- fixed now, can you please reapprove?

**Pull request merged**